src/qt/rpcconsole.cpp

```diff
@@ -1081,15 +1079,13 @@ void RPCConsole::peerLayoutChanged()
 
 void RPCConsole::updateDetailWidget()
 {
-    QModelIndexList selected_rows;
-    auto selection_model = ui->peerWidget->selectionModel();
-    if (selection_model) selected_rows = selection_model->selectedRows();
-    if (!clientModel || !clientModel->getPeerTableModel() || selected_rows.size() != 1) {
+    const auto selected_peers = GUIUtil::getEntryData(ui->peerWidget, PeerTableModel::NetNodeId);
+    if (!clientModel || !clientModel->getPeerTableModel() || selected_peers.size() != 1) {
         ui->detailWidget->hide();
         ui->peerHeading->setText(tr("Select a peer to view detailed information."));
         return;
     }
-    const CNodeCombinedStats *stats = clientModel->getPeerTableModel()->getNodeStats(selected_rows.first().row());
+    const auto stats = selected_peers.first().data(PeerTableModel::StatsRole).value<CNodeCombinedStats*>();
```

Comment:
6e27ee8
```suggestion
    const CNodeCombinedStats* stats{selected_peers.first().data(PeerTableModel::StatsRole).value<CNodeCombinedStats*>()};
```

Reply:
Type is already in `.value<>()`, IIRC `auto` can be used when the type is clear from the statement.

Reply:
True

Reply:
Marked as "resolved" without changes.